rockcraft.yaml

```
@@ -1,71 +1,65 @@
+  - JAVA_HOME: /usr/lib/jvm/java-17-openjdk-amd64
+  - ZOO_LOG_DIR: /var/log/zookeeper
+  - ZOO_LOG4J_PROP: INFO,CONSOLE,ROLLINGFILE
+  - ZOO_LOG_FILE: zookeeper.log
+  - ZOOCFGDIR: /etc/zookeeper
 
 parts:
   zookeeper:
     plugin: nil
     stage-packages:
-      - openjdk-8-jre-headless
+      - openjdk-17-jre-headless
+      - libpsl5
+      - curl
     build-environment:
-      - JAVA_HOME: /usr/lib/jvm/java-8-openjdk-amd64/
+      - JAVA_HOME: /usr/lib/jvm/java-17-openjdk-amd64
     stage-snaps:
-      - zookeeper/3.6/edge
-    override-build: |
-      rm -rf usr/
-    organize:
-      LICENSE.txt : opt/zookeeper/LICENSE.txt
-      NOTICE.txt : opt/zookeeper/NOTICE.txt
-      README.md : opt/zookeeper/README.md
-      README_packaging.md : opt/zookeeper/README_packaging.md
-      bin : opt/zookeeper/bin
-      conf : opt/zookeeper/conf
-      docs : opt/zookeeper/docs
-      etc : opt/zookeeper/etc
-      lib : opt/zookeeper/lib
-      with_config.sh : opt/zookeeper/with_config.sh
-      jmx_prometheus_javaagent.jar : opt/zookeeper/jmx_prometheus_javaagent.jar
-      jmx_prometheus.yaml :  opt/zookeeper/conf/jmx_prometheus.yaml
-  java:
-    plugin: nil
-    after: [zookeeper]
-    stage-packages:
-      - openjdk-8-jre-headless
-    build-environment:
-      - JAVA_HOME: /usr/lib/jvm/java-8-openjdk-amd64/
+      - charmed-zookeeper/3/edge
     override-build: |
-      ln -s /usr/lib/jvm/java-8-openjdk-amd64/bin/java \
+      ln -s /usr/lib/jvm/java-17-openjdk-amd64/bin/java \
         $CRAFT_PART_INSTALL/usr/bin/java
-      ln -s /usr/lib/jvm/java-8-openjdk-amd64/bin/keytool \
+      ln -s /usr/lib/jvm/java-17-openjdk-amd64/bin/keytool \
         $CRAFT_PART_INSTALL/usr/bin/keytool
+
+      mkdir -p $CRAFT_PART_INSTALL/var/lib/pebble/default/
+      mkdir -p $CRAFT_PART_INSTALL/var/lib/zookeeper/
+      mkdir -p $CRAFT_PART_INSTALL/var/log/zookeeper/
+      mkdir -p $CRAFT_PART_INSTALL/opt/zookeeper/
+      mkdir -p $CRAFT_PART_INSTALL/etc/zookeeper/
     override-prime: |
       craftctl default
-      rm -vf usr/lib/jvm/java-8-openjdk-amd64/jre/lib/security/cacerts
-
+      rm -vf usr/lib/jvm/java-17-openjdk-amd64/lib/security/cacerts
   non-root-user:
-        plugin: nil
-        after: [java]
-        overlay-script: |
-          # Create a user in the $CRAFT_OVERLAY chroot
-          groupadd -R $CRAFT_OVERLAY -g 1000 zookeeper
-          useradd -R $CRAFT_OVERLAY -M -r -g zookeeper -u 1000 zookeeper
-        override-prime: |
-          craftctl default
-          # Give permission to the required folder
-          chown -R 1000:1000 opt/zookeeper
+    plugin: nil
+    after: [zookeeper]
+    overlay-script: |
+      # Create a user in the $CRAFT_OVERLAY chroot
+      groupadd -R $CRAFT_OVERLAY -g 1000 zookeeper
+      useradd -R $CRAFT_OVERLAY -M -r -g zookeeper -u 1000 zookeeper
+    override-prime: |
+      craftctl default
+      # Give permissions to the required folders
+      chown -R 1000:1000 opt/zookeeper
```

it is simpler create dirs ONE bash command (not two separate `mkdir` and `chown` commands)
```suggestion
      install -d -o 1000 -g 1000 -m 755 \
          opt/zookeeper \
          var/lib/zookeeper \
          var/log/zookeeper \
          etc/zookeeper
```